### ISSUE TITLE
Other Feature weightings

### ISSUE THREAD

**Issue by rth** [FreeDiscovery/FreeDiscovery#29]
In some cases, features weighting such as Delta TFIDF  and BM25 were reported to produce better binary categorization scores that TFIDF weights, cf https://www.mail-archive.com/scikit-learn-general@lists.sourceforge.net/msg11328.html 

 - BM25 is currently an open PR at scikit-learn
 - Delta TDFIDF  was reported to produce better binary categorization scores by [Martineau et al (2009)](http://www.vcl.salk.edu/~spatel/DeltaTFIDFPaper.pdf)
  - A Supervised tf (tf-chi², tf-rf) implementation can also be found here: https://gist.github.com/larsmans/239fecd3fc6b49e50da9

Not sure if this would be useful in FreeDiscovery, and it's definitely low priority, but just opening this issue (with useful links) for future reference.

**Comment by rth** [comment]
@eugene-yang You mentioned you would be interested in custom feature extraction options. 

For instance, I'm wondering if we could optionally allow feature extraction with gensim's [word2vec](https://radimrehurek.com/gensim/models/word2vec.html) and then use mostly the same methods that currently operate in the LSI space without modifying  the API too much.  

[spaCy](https://spacy.io/) might also be worth considering.

The main constant here is that the REST API should be fixed for v1.0 while we can more easily modify python code since a number of functions are not exposed in the public API...

What do you think, @eugene-yang ?

**Comment by eugene-yang** [comment]
I think use w2v is a good option.

Is it possible to create make the REST API more like a proxy instance for the python API? In that case you will not need to update the REST API interface every time we have a new method in the python APIs.

**Comment by rth** [comment]
> Is it possible to create make the REST API more like a proxy instance for the python API? In that case you will not need to update the REST API interface every time we have a new method in the python APIs.

I agree completely. The only question is which input parameters the REST API v1.0 should have to allow w2v some time later in the future...
